### PR TITLE
luks: add extraArgs{Format,Open}

### PR DESCRIPTION
**Pull request by nazarewk**
There are number of fixes I needed to make to support my use case, they're brewing up here.

### CODE REVIEW

File: types.nix [hunk 1]
```diff
@@ -1,9 +1,9 @@
 { lib }:
 with lib;
 with builtins;
-
-rec {
-
+let
+  safeZpoolName = pool: builtins.replaceStrings [ "-" ] [ "_" ] pool;
+in rec {
```
Comment (Lassulus) [inline types.nix:6]: hmm, I would prefer to abort on a invalid/unsafe zpool name rather than to replace it silently. replacing could lead to conflicts

Comment (nazarewk) [reply types.nix:6]: it is for specific purpose of creating an environment variable name, could make it clearer

Comment (Lassulus) [reply types.nix]: could they still conflict on similar zpool names? like my-pool and my_pool? maybe we don't care about that?

Comment (nazarewk) [reply types.nix]: this should solve it once and for good https://github.com/nix-community/disko/pull/112/commits/9eeafb271583e4fbe7c05580cf9debcb7554aea0

Comment (Lassulus) [reply types.nix]: I guess we use now tmpfiles for the pool names, not sure if this change is still relevant

**Comment by nazarewk** [comment]
I've rebased the branch, but will test it and finish up the PR after setting up second laptop sometime next week

**Comment by Lassulus** [comment]
I guess this is now partially obsolete due to https://github.com/nix-community/disko/pull/186

**Comment by nazarewk** [comment]
this was addressed by #186 and #211

**Pull request closed without merging**